lib/secure_sockets/portable/freertos_plus_tcp/aws_secure_sockets.c

```diff
@@ -645,20 +630,24 @@ static CK_RV prvSocketsGetCryptoSession( CK_SESSION_HANDLE * pxSession,
 uint32_t ulRand( void )
 {
     CK_RV xResult = 0;
+    SemaphoreHandle_t xSessionLock = NULL;
     CK_SESSION_HANDLE xPkcs11Session = 0;
     CK_FUNCTION_LIST_PTR pxPkcs11FunctionList = NULL;
     uint32_t ulRandomValue = 0;
 
-    xResult = prvSocketsGetCryptoSession( &xPkcs11Session,
+    xResult = prvSocketsGetCryptoSession( &xSessionLock,
+                                          &xPkcs11Session,
                                           &pxPkcs11FunctionList );
 
     if( 0 == xResult )
     {
         /* Request a sequence of cryptographically random byte values using
          * PKCS#11. */
+        xSemaphoreTake( xSessionLock, portMAX_DELAY );
```

Comment:
These semaphore lock/give can be deleted -- locking the session for a GenerateRandom call should not be necessary, since the PKCS #11 implementation should protect calls to GenerateRandom already.

Reply:
Fixed.